packages/gateway/src/errors/as-gateway-error.ts

```diff
@@ -2,7 +2,10 @@ import { APICallError } from '@ai-sdk/provider';
 import { extractApiCallResponse, GatewayError } from '.';
 import { createGatewayErrorFromResponse } from './create-gateway-error';
 
-export function asGatewayError(error: unknown) {
+export function asGatewayError(
+  error: unknown,
+  authMethod?: 'api-key' | 'oidc',
```

Comment:
this fn is intended to be error-subtype agnostic, wondering if we can keep it that way somehow, need to look at control flow more closely with you